chia/wallet/puzzles/clawback/drivers.py

```diff
@@ -0,0 +1,156 @@
+    merkle_tree: Tuple[bytes32, Dict[bytes32, Tuple[int, List[bytes32]]]], puzzle_hash: bytes32
+) -> Program:
+    """
+    To spend a p2_1_of_n clawback we recreate the full merkle tree: (root, dict_of_proofs)
+    The required proof is then selected from dict_of_proofs based on the puzzle_hash of the puzzle we
+    want to execute
+    Returns a proof: (int, List[bytes32]) which can be provided to the p2_1_of_n solution
+    """
+    proof: Program = Program.to(merkle_tree[1][puzzle_hash])
+    return proof
+
+
+def create_merkle_puzzle(timelock: uint64, sender_ph: bytes32, recipient_ph: bytes32) -> Program:
+    merkle_tree = create_clawback_merkle_tree(timelock, sender_ph, recipient_ph)
+    puzzle: Program = P2_1_OF_N.curry(merkle_tree[0])
+    return puzzle
+
+
+def create_merkle_solution(
+    timelock: uint64,
+    sender_ph: bytes32,
+    recipient_ph: bytes32,
+    inner_puzzle: Program,
+    inner_solution: Program,
+) -> Program:
+    """
+    Recreates the full merkle tree of a p2_1_of_n clawback coin. It uses the timelock and each party's
+    puzhash to create the tree.
+    The provided inner puzzle must hash to match either the sender or recipient puzhash
+    If it's the sender, then create the clawback solution. If it's the recipient then create the claim
+    solution.
+    Returns a program which is the solution to a p2_1_of_n clawback.
+    """
+    merkle_tree = create_clawback_merkle_tree(timelock, sender_ph, recipient_ph)
+    inner_puzzle_hash = inner_puzzle.get_tree_hash()
+    if inner_puzzle_hash == sender_ph:
+        cb_inner_puz = create_p2_puzzle_hash_puzzle(sender_ph)
+        merkle_proof = create_merkle_proof(merkle_tree, cb_inner_puz.get_tree_hash())
+        cb_inner_solution = create_p2_puzzle_hash_solution(inner_puzzle, inner_solution)
+    elif inner_puzzle_hash == recipient_ph:
+        condition = [80, timelock]
+        cb_inner_puz = create_augmented_cond_puzzle(condition, inner_puzzle)
+        merkle_proof = create_merkle_proof(merkle_tree, cb_inner_puz.get_tree_hash())
+        cb_inner_solution = create_augmented_cond_solution(inner_solution)
+    else:
+        raise ValueError("Invalid Clawback inner puzzle.")
+    solution: Program = Program.to([merkle_proof, cb_inner_puz, cb_inner_solution])
+    return solution
+
+
+def match_clawback_puzzle(inner_puzzle: Program, inner_solution: Program, max_cost: int) -> Optional[ClawbackMetadata]:
+    # Check if the inner puzzle is a P2 puzzle
+    if MOD != uncurry_puzzle(inner_puzzle).mod:
+        return None
+    # Fetch Remark condition
+    conditions = conditions_for_solution(
+        inner_puzzle,
+        inner_solution,
+        max_cost,
+    )
+    if conditions is not None:
+        for condition in conditions:
+            if (
+                condition.opcode == ConditionOpcode.REMARK
+                and len(condition.vars) == 2
+                and condition.vars[0] == bytes(PuzzleDecoratorType.CLAWBACK.name, "utf-8")
```

Comment:
it seems suspicious to use a (utf-8) string to identify the clawback metadata. Shouldn't this just be some unique integer? this ends up stored on chain, so I would think we want it to be as small as possible

Reply:
We can use a magic number to represent this string. Just make it as unique as possible.